### PR TITLE
Lw/shader keywords stripping

### PR DESCRIPTION
**Pull request by phi-lira**
- Added C# shader stripping rules:
     -- Standard Lit shader variant count can be reduced by 96.875%. 
     -- Meta pass stripped from build
     -- ScreenSpace shadow and Shadow passes are stripped from build if shadows are disabled.
     -- All shaders with Debug in their name are stripped from build.
     -- HDRenderPipeline shaders are stripped from build.
  - Local lights now support soft shadows and work on mobile
     -- _LOCAL_SHADOWS_ENABLED and _SOFT_SHADOWS keywords were added.
  - Pipeline Asset now lets configure different settings for Directional and Local shadows.

### CODE REVIEW

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/Editor/LightweightAssetEditor.cs [hunk 1]
```diff
@@ -28,25 +28,29 @@ internal class Styles
...
             public static GUIContent opaqueDownsampling = new GUIContent("Opaque Downsampling", "The downsampling method that is used for the opaque texture");
+            public static GUIContent hdrContent = new GUIContent("HDR", "Controls the global HDR settings.");
+            public static GUIContent msaaContent = new GUIContent("Anti Aliasing (MSAA)", "Controls the global anti aliasing settings.");
 
-            public static GUIContent shadowType = new GUIContent("Type",
-                    "Global shadow settings. Options are NO_SHADOW, HARD_SHADOWS and SOFT_SHADOWS.");
 
-            public static GUIContent shadowDistante = new GUIContent("Distance", "Max shadow rendering distance.");
+            public static GUIContent supportsSoftShadows = new GUIContent("Soft Shadows", "If enabled pipeline will perform shadow filtering. Otherwise all shadows will only perform a single shadow sample.");
+            public static GUIContent supportsDirectionalShadows = new GUIContent("Directional Shadows", "If disabled all directional lights won't cast shadows.");
```
Comment (aleks01010101) [inline ScriptableRenderPipeline/LightweightPipeline/LWRP/Editor/LightweightAssetEditor.cs]: Probably better to keep those in sync. E.g. if one says "if enabled", others should also say "if enabled".

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/Editor/LightweightAssetEditor.cs [hunk 2]
```diff
@@ -151,21 +160,59 @@ public override void OnInspectorGUI()
             EditorGUI.indentLevel--;
             EditorGUILayout.Space();
             EditorGUILayout.Space();
+        }
+
+        void DrawShadowSettings()
+        {
+            bool directionalShadows = false;
```
Comment (aleks01010101) [inline ScriptableRenderPipeline/LightweightPipeline/LWRP/Editor/LightweightAssetEditor.cs]: I'd declare those when they're first used

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightPipeline.cs [hunk 3]
```diff
@@ -957,23 +958,20 @@ private void SetShaderKeywords(CommandBuffer cmd, ref LightData lightData)
...
-
-            // TODO: Currently we are shading same keyword for directional and local lights
-            // When we have ability to strip keywords we can use the one below instead
-            //CoreUtils.SetKeyword(cmd, "_SHADOWS_ENABLED", m_ShadowPass.HasDirectionalShadowmap);
-            //CoreUtils.SetKeyword(cmd, "_LOCAL_SHADOWS_ENABLED", m_ShadowPass.HasLocalLightsShadowmap);
-
+            CoreUtils.SetKeyword(cmd, LightweightKeywords.AdditionalLightsText, lightData.totalAdditionalLightsCount > 0);
+            CoreUtils.SetKeyword(cmd, LightweightKeywords.MixedLightingSubtractiveText, m_MixedLightingSetup == MixedLightingSetup.Subtractive);
+            CoreUtils.SetKeyword(cmd, LightweightKeywords.VertexLightsText, vertexLightsCount > 0);
             //TIM: Not used in shader for V1 to reduce keywords
```
Comment (aleks01010101) [inline ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightPipeline.cs]: This comment can probably go

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightPipelineCore.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        public static PipelineCapabilities GetPipelineCapabilities()
+        {
+            return s_PipelineCapabilities;
+        }
+
+        static void SetPipelineCapabilities(LightweightPipelineAsset pipelineAsset)
+        {
+            s_PipelineCapabilities = 0U;
+
+            if (pipelineAsset.MaxPixelLights < 1 && !pipelineAsset.SupportsVertexLight)
```
Comment (aleks01010101) [inline ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightPipelineCore.cs]: Shouldn't it be `> 1`?

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightShadowPass.cs [hunk 5]
```diff
@@ -32,17 +32,18 @@ public static ShadowSettings Default
                 if (defaultShadowSettings == null)
                 {
                     defaultShadowSettings = new ShadowSettings();
-                    defaultShadowSettings.directionalShadowQuality = LightShadows.None;
+                    defaultShadowSettings.supportsDirectionalShadows = true;
                     defaultShadowSettings.screenSpace = true;
                     defaultShadowSettings.directionalShadowAtlasHeight = defaultShadowSettings.directionalShadowAtlasWidth = 2048;
-                    defaultShadowSettings.localLightsShadowQuality = LightShadows.None;
+                    defaultShadowSettings.directionalLightCascadeCount = 1;
+                    defaultShadowSettings.directionalLightCascades = new Vector3(0.067f, 0.2f, 0.467f);
```
Comment (aleks01010101) [inline ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightShadowPass.cs]: Why did the default change?

Comment (phi-lira) [reply ScriptableRenderPipeline/LightweightPipeline/LWRP/LightweightShadowPass.cs]: I just matched the default with the inspector default. Should probably have a default constant for this.

File: ScriptableRenderPipeline/LightweightPipeline/LWRP/Data/LightweightPipelineAsset.cs [hunk 6]
```diff
@@ -289,13 +285,25 @@ public int CascadeCount
...
+        public bool IsLocalShadowsSupported
+        {
+            get { return m_LocalShadowsSupported; }
+        }
+
+        public int LocalShadowAtlasResolution
+        {
+            get { return (int)m_LocalShadowsAtlasResolution; }
+        }
+        public bool IsSoftShadowsSupported
```
[the remark this answers is not in the document]
Comment (phi-lira) [reply ScriptableRenderPipeline/LightweightPipeline/LWRP/Data/LightweightPipelineAsset.cs]: The convention leaves optional to use Is/Has (doesn't mention Are). I've changed it to Supports* instead. I should say that this file is not yet in the new C# convention which uses camelCase on properties (Exception to FDG).

**Pull request merged**